src/framework/chain_normalizers.ini

```diff
@@ -6,4 +6,5 @@
 # The names of the services on the right dictate the preference used (if unavailable
 # the second and third are applied as applicable).
 
+deinterlacer=deinterlace,avdeinterlace
```

Comment:
I was just thinking about adding a movit-based deinterlacer when I though to myself filter or link? It turns out to be a link because it being YADIF-based takes 5 fields for input buffers and can be driven to frame double. I am going to do something else while waiting for this to merge, and let me know if you want to tackle that next as I think it would a good next step.

But help me understand the priority between filter and link for the same purpose. Of course, if you are not using a chain then a link is not used. The chain wraps a producer that has normalizing filters attached. So how does the smarter link-based resampler get used if there is a filter-based resampler upstream that already did the work? This along with chains and links in general needs a document in the miscellaneous section of https://mltframework.org/docs/. Do not need to go into huge depth but cover the basic concepts, purposes, and ways they get included including API, melt, and XML.

Reply:
> I was just thinking about adding a movit-based deinterlacer when I though to myself filter or link? It turns out to be a link because it being YADIF-based takes 5 fields for input buffers and can be driven to frame double. I am going to do something else while waiting for this to merge, and let me know if you want to tackle that next as I think it would a good next step.

I am happy to work on a movit-based deinterlacer link. But first I would like to finish my "forward speed filter" work that I have in progress.

> But help me understand the priority between filter and link for the same purpose. Of course, if you are not using a chain then a link is not used. 

This is the "magic" that allows a link to tell a downstream normalizing filter that it does not need to do anything:
https://github.com/bmatherly/mlt/blob/deinterlace_link/src/modules/xine/link_deinterlace.c#L42
Essentially, you have to manipulate the frame and get_image() parameters so that the downstream normalizers do not think that anything needs to be done.
This unfortunately means that normalization is needed after the deinterlacer link. This is a work-around for now:
https://github.com/bmatherly/mlt/blob/deinterlace_link/src/modules/xine/link_deinterlace.c#L268
But eventually I will make normalizer links for all normalization types and then we will not need normalizers on the encapsulated producer.

> This along with chains and links in general needs a document in the miscellaneous section of https://mltframework.org/docs/. Do not need to go into huge depth but cover the basic concepts, purposes, and ways they get included including API, melt, and XML.

That is a good idea. I will add that to my todo list.